### ISSUE TITLE
terminal -- control code corruption

### ISSUE THREAD

**Issue by williamstein** [sagemathinc/cocalc#1269]
We just haven't solved this problem yet.

This is perhaps something that should be done as part of the terminal react rewrite... though maybe not totally (it's mostly orthogonal to that).

**Comment by williamstein** [comment]
> Jennifer Balakrishnan: "Currently for me, if I have either one of these two terminals open in
addition to any other file in any other SMC project, then the `;1R` happens each time I switch back to the problem terminals:"

**Comment by haraldschilly** [comment]
here is a screenshot ... just running some doctests and I wasn't even at the computer touching anything. i.e. it started all by its own (no resizing, etc.)
![smc-terminal-1r](https://cloud.githubusercontent.com/assets/207405/21982038/b623545a-dbea-11e6-9264-8679f785f02a.png)

**Comment by williamstein** [comment]
Changed the subject heading since the single-user case has the same issues.

**Comment by haraldschilly** [comment]
another screenshot. closing/re-opening the terminal tab always solves it. I'll again try to find out what's going on ...

![terminal-control-chars](https://cloud.githubusercontent.com/assets/207405/22065214/85905312-dd87-11e6-8677-cd4a53e4daf1.png)

**Comment by williamstein** [comment]
Also, when doing this, move all resize-related information to a separate communication channel, which would solve a longstanding bug involving codes like " "4;34;141t;..."" showing up when resizing...

**Comment by williamstein** [comment]
I never saw this since I use tmux all the time.  It's really really easy to reproduce this by not using tmux.   I see it constantly when not using tmux.

**Comment by DrXyzzy** [comment]
I was not able to reproduce the problem with 3 differently-sized browser windows open on the same .term, two from FF and one from Chrome. Switching between windows and typing & resizing did not cause display problems or introduce extraneous characters.

I did notice that if I type `open foo.tex` in one .term window, all three terminal sessions switch to the latex editor view.

**Comment by williamstein** [comment]
> I was not able to reproduce the problem

I still find it impossible to **not** reproduce this problem within seconds...

> I did notice that if I type open foo.tex in one .term window, all three terminal sessions switch to the latex editor view.

This is currently the expected behavior, since open just sends some codes which are "rendered" by doing the open... if that terminal is focused (and not otherwise).

**Comment by williamstein** [comment]
I think this is now fixed...

**Comment by williamstein** [comment]
Process: Um, I finally fixed that terminal corruption bug.  It was easy and had nothing directly to do with resizing the terminal!  But was hard to reproduce in smc-in-smc... but easy in Docker, so I fixed it by doing dev on my local Docker container.  Problem was just term.js send output to the backned **while** re-rendering past terminal history after a resize/reconnect/refresh event; solution - don't send output during re-render.
